learn/guides/organizing-ballerina-code.md

```diff
@@ -0,0 +1,144 @@
+- /learn/user-guide/ballerina-packages/organizing-ballerina-code/
+- /learn/organizing-ballerina-code
+- /learn/user-guide/structuring-ballerina-code/
+- /learn/user-guide/structuring-ballerina-code
+- /learn/user-guide/ballerina-packages/
+- /learn/user-guide/ballerina-packages
+---
+
+## Package Structure
+
+Writing code in an organized manner from the beginning of the project is important for the lifecycle of the project and its maintainability in the long run. Organized code will make it easy to extend and improve your project over time. Ballerina project structure makes it easy to write clean code by eliminating repetitions, writing reusable code, adding new features without changing the existing code etc. To achieve this, Ballerina has the concept of Packages and modules. 
+
+Ballerina code is organized in a single shareable unit called a Package. Package is a collection of modules and a module is a collection of Ballerina source files, test files and resources. A package should contain at least one module which is called the default module. Each module has its own directory which is used to organize source files, test files and resources.
+
+A package is a collection of related modules that are versioned and distributed as a single unit. It is common in small projects to have only one module (default) in a package. As a result, the default module’s content is placed directly in the root of the package directory.
+
+`bal new` command creates the Package with the default module. This will generate the `Ballerina.toml` which identifies a directory as a package and will additionally generate a sample source file in the default module.
+
+### Creating your first Ballerina package
+
+You can create a Ballerina package with the `bal new` command as follows. 
+
+```bash
+bal new hello_world
+```
+
+This will create a new Ballerina package with a main function. 
+
+The `bal new` command generates the following file structure.
+
+```bash
+cd hello_world
+tree .
+.
+├── Ballerina.toml
+└── main.bal
+    
+0 directories, 2 files
+```
+
+You may also try creating a service or a lib package instead of a main function. 
+
+#### Creating a Ballerina service package
+
+```bash
+bal new -t service hello_service
+```
+
+This will create a Ballerina source containing a service declaration with Ballerina tests to test the service. The following file structure will be generated with the service template.  
+
+```bash
+cd hello_service
+.
+├── Ballerina.toml
+├── service.bal
+└── tests
+    └── service_test.bal
+
+1 directory, 3 files
+```
+
+#### Creating a library Package
+
+```bash
+bal new -t lib hello_lib
+```
+
+This will create a Ballerina source file containing a function that prints `Hello, world!` along with a test file to test the function. 
+Additionally, it will also create the `Package.md` file which is required to [publish a package to Ballerina Central]((/learn/user-guide/publishing-packages-to-ballerina-central).
```

Comment:
```suggestion
Additionally, it will also create the `Package.md` file, which is required to [publish a package to Ballerina Central]((/learn/user-guide/publishing-packages-to-ballerina-central).
```